src/zoid/buttons/util.js

```diff
@@ -357,3 +362,59 @@ export function getButtonSize(
     }
   }
 }
+
+export const getModal: (
+  clientID: string,
+  merchantID: $ReadOnlyArray<string> | void
+) => Object = memoize((clientID, merchantID) => {
+  const namespace = getNamespace();
+
+  if (!window[namespace]?.MessagesModal) {
+    const modalBundleUrl = () => {
+      let envPiece;
+      switch (getEnv()) {
+        case "local":
+        case "test":
+        case "stage":
+          envPiece = "stage";
+          break;
+        case "sandbox":
+          envPiece = "sandbox";
+          break;
+        case "production":
+        default:
+          envPiece = "js";
+      }
+      return `https://www.paypalobjects.com/upstream/bizcomponents/${envPiece}/modal.js`;
+    };
+
+    try {
+      // eslint-disable-next-line no-restricted-globals
+      return new Promise(() => {
```

Comment:
If we return the Promise then the modal creation code below will never execute since the function is memoized. We will need to `await` the promise instead so that it becomes just another step.